src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2395,7 +2436,7 @@ private function registerHttpClientConfiguration(array $config, ContainerBuilder
         }
 
         if ($responseFactoryId = $config['mock_response_factory'] ?? null) {
-            $container->register($httpClientId.'.mock_client', MockHttpClient::class)
+            $container->register('http_client.mock_client', MockHttpClient::class)
```

Comment:
I don't understand why this id needs to be dynamic :thinking: I hardcoded it instead of making it dynamic in the failing test.

Reply:
ppl might configure a different mock clients per named clients in functional tests I suppose?

Reply:
We indeed must have separate decorators for each http client service (the same definition cannot decorate multiple services)

Reply:
I'll just keep it as it is and update the test because it's unrelated to this PR.